apollo-client/src/main/java/com/ctrip/framework/apollo/util/ConfigUtil.java

```diff
@@ -410,23 +412,38 @@ public boolean isPropertyNamesCacheEnabled() {
     return propertyNamesCacheEnabled;
   }
 
+  public boolean isPropertyFileCacheEnabled() {
+    return propertyFileCacheEnabled;
+  }
+
   private void initPropertyNamesCacheEnabled() {
-    String propertyName = ApolloClientSystemConsts.APOLLO_PROPERTY_NAMES_CACHE_ENABLE;
-    String propertyEnvName = ApolloClientSystemConsts.APOLLO_PROPERTY_NAMES_CACHE_ENABLE_ENVIRONMENT_VARIABLES;
+    propertyNamesCacheEnabled = getPropertyBoolean(ApolloClientSystemConsts.APOLLO_PROPERTY_NAMES_CACHE_ENABLE,
+            ApolloClientSystemConsts.APOLLO_PROPERTY_NAMES_CACHE_ENABLE_ENVIRONMENT_VARIABLES,
+            "false");
+  }
+
+  private void initPropertyFileCacheEnabled() {
+    propertyFileCacheEnabled = getPropertyBoolean(ApolloClientSystemConsts.APOLLO_FILE_CACHE_ENABLE,
+            ApolloClientSystemConsts.APOLLO_FILE_CACHE_ENABLE_ENVIRONMENT_VARIABLES,
+            "true");
+  }
+
+  private boolean getPropertyBoolean(String propertyName, String envName, String defaultVal) {
```

Comment:
the default value could simply be `boolean`?
```suggestion
  private boolean getPropertyBoolean(String propertyName, String envName, boolean defaultVal) {
```

Reply:
call `Foundation.app().getProperty()` must have a `String defaultVal`. I can change this to `boolean`, but it will looks ugly for doing many `String` `Boolean` convert. How about support `getBooleanProperty` in `Foundation.app()`? Or get without `String defaultVal`

Reply:
The return type of getPropertyBoolean is boolean, so the default value is better keep type same as it. that will let method more understandable. 

We can keep the complex inside the method getPropertyBoolean, i.e convert boolean to String like follow?

```java
private boolean getPropertyBoolean(String propertyName, String envName, boolean defaultVal) {
    // ...
    Foundation.app().getProperty(propertyName, String.valueOf(defaultVal));
    // ...
}
```

Reply:
Yes we can, But if support `getBooleanProperty` or `getProperty` without default value, it will looks more graceful

Reply:
we could simply pass null as the default value to read from Foundation.app()
```java
Foundation.app().getProperty(propertyName, null);
```

Reply:
fixed